src/azul/service/manifest.py

```diff
@@ -19,7 +27,23 @@ def encode_params(self, params: dict) -> str:
     def decode_params(self, token: str) -> dict:
         return json.loads(base64.urlsafe_b64decode(token).decode('utf-8'))
 
-    def start_manifest_generation(self, filters: dict, execution_id: str):
+    def run(self, token, retry_url, filters='{}'):
+        """
+        If a token is not empty it is decoded for a execution_id. The status of the state machine with
+        this id is reported back.
```

Comment:
This is both too much and too little detail. If you go to the trouble of writing documentation it should be precise.

> If token is None, start a manifest generation process and returns its status. Otherwise return the status of the manifest generation process represented by the token.

I wish token would be packed and unpacked in a single place.

Reply:
Changed to your suggestion.